converter/dicom-cast/src/Microsoft.Health.DicomCast.Core/Features/Worker/FhirTransaction/Observation/ObservationParser.cs

```diff
@@ -0,0 +1,354 @@
+                }
+            }
+
+            Identifier studyInstanceIdentifier = ImagingStudyIdentifierUtility.CreateIdentifier(studyInstanceUid);
+            observation.Identifier.Add(studyInstanceIdentifier);
+
+            // Try to get accession number from report first then tag; ignore if it is not present it is not a required identifier.
+            var accessionNumber = report.Get<string>(ObservationConstants.AccessionNumber, "");
+            if (string.IsNullOrEmpty(accessionNumber))
+            {
+                dataset.TryGetSingleValue(DicomTag.AccessionNumber, out accessionNumber);
+            }
+
+            if (!string.IsNullOrEmpty(accessionNumber))
+            {
+                var identifier = new Identifier
+                {
+                    Value = accessionNumber,
+                    Type = new CodeableConcept("http://terminology.hl7.org/CodeSystem/v2-0203", "ACSN")
+                };
+                observation.Identifier.Add(identifier);
+            }
+
+            // Add all structured report information
+            ApplyDicomTransforms(observation, dataset, new Collection<DicomCodeItem>()
+            {
+                ObservationConstants.DoseRpTotal,
+                ObservationConstants.AccumulatedAverageGlandularDose,
+                ObservationConstants.DoseAreaProductTotal,
+                ObservationConstants.FluoroDoseAreaProductTotal,
+                ObservationConstants.AcquisitionDoseAreaProductTotal,
+                ObservationConstants.TotalFluoroTime,
+                ObservationConstants.TotalNumberOfRadiographicFrames,
+                ObservationConstants.AdministeredActivity,
+                ObservationConstants.CtDoseLengthProductTotal,
+                ObservationConstants.TotalNumberOfIrradiationEvents,
+                ObservationConstants.MeanCtdIvol,
+                ObservationConstants.RadiopharmaceuticalAgent,
+                ObservationConstants.RadiopharmaceuticalVolume,
+                ObservationConstants.Radionuclide,
+                ObservationConstants.RouteOfAdministration,
+            });
+
+            return observation;
+        }
+
+        /// <summary>
+        /// Creates an Observation "Irradiation Event" based on the spec outlined at:
+        /// https://confluence.hl7.org/display/IMIN/Radiation+Dose+Summary+for+Diagnostic+Procedures+on+FHIR#RadiationDoseSummaryforDiagnosticProceduresonFHIR-DoseSummaryObservationProfile
+        /// </summary>
+        /// <param name="dataset">The dataset to parse</param>
+        /// <param name="patientRef">Patient which the Irradiation Event is a subject of</param>
+        private static Observation CreateIrradiationEvent(DicomDataset dataset, ResourceReference patientRef)
+        {
+            var report = new DicomStructuredReport(dataset);
+            // create the observation
+            var observation = new Observation
```

Comment:
Looks like the following fields are marked as cardinality of `1..1` in radiation event:

- BodySite
- effective[x]